pkg/kubectl/sorting_printer.go

```diff
@@ -185,16 +186,37 @@ func isLess(i, j reflect.Value) (bool, error) {
 
 func (r *RuntimeSort) Less(i, j int) bool {
 	iObj := r.objs[i]
+	iElem := reflect.ValueOf(iObj).Elem()
 	jObj := r.objs[j]
+	jElem := reflect.ValueOf(jObj).Elem()
+	if r.field == "" && iElem.Type().Name() == "ReplicaSet" && jElem.Type().Name() == "ReplicaSet" {
```

Comment:
This is a huge encapsulatin violation - we should come up with a clean way to solve this, rather than hacking it in here.